stl/inc/flat_set

```diff
@@ -153,6 +153,27 @@ public:
     _Base_flat_set(_Tsorted _Tsort, initializer_list<_Kty> _Ilist, const _Alloc& _Al)
         : _Base_flat_set(_Tsort, container_type(_Ilist, _Al)) {}
 
+    _Base_flat_set(const _Base_flat_set&) = default;
+    _Base_flat_set(_Base_flat_set&& _Other)
+        : _Mycont(_STD move(_Other).extract()), _Mycomp(_Other._Mycomp) // intentionally copy comparator, see LWG-2227
+    {}
+
+    _Base_flat_set& operator=(const _Base_flat_set& _Other) {
+        _Clear_guard<_Base_flat_set> _Guard{this};

```

Comment:
`_Guard` is needed when:
1. the copy assignment fails and cannot provide strong-guarantee, or
2. `_Mycomp` is different and throws during copy.

Reply:
Though not necessary, I think it looks strange that move-assignment has address checking, while this doesn't have :|